src/handlers/setDependencyVersionHandler.ts

```diff
@@ -0,0 +1,198 @@
+    } else if (targetNode.tag === XmlTagName.Project) {
+        insertPosition = currentDocument.positionAt(targetNode.contentEnd);
+        targetText = constructDepedencyManagementNode(gid, aid, version, baseIndent, indent, eol);
+    } else {
+        return;
+    }
+
+    const edit: vscode.WorkspaceEdit = new vscode.WorkspaceEdit();
+    if (deleteNode) { // the version of ${gid:aid} dependency node already imported should be deleted
+        const versionItem: ElementNode | undefined = deleteNode.children?.find(node => node.tag === XmlTagName.Version);
+        if (versionItem) {
+            if (versionItem.contentStart === undefined || versionItem.contentEnd === undefined) {
+                throw new UserError("Invalid target XML node to delete.");
+            }
+            const versionString: string = "<version>";
+            const start: number = versionItem.contentStart - versionString.length;
+            const end: number = versionItem.contentEnd + versionString.length + 1;
+            const range = new vscode.Range(currentDocument.positionAt(start), currentDocument.positionAt(end));
+            edit.delete(currentDocument.uri, range);
+        }
+    }
+    if (dependencyNodeInManagement) { // ${gid:aid} dependency node that already exists in <dependencyManagement> shoule be deleted
+        if (dependencyNodeInManagement.contentStart === undefined || dependencyNodeInManagement.contentEnd === undefined) {
+            throw new UserError("Invalid target XML node to delete.");
+        }
+        const dependencyString: string = "<dependency>";
+        const start = dependencyNodeInManagement.contentStart - dependencyString.length;
+        const end = dependencyNodeInManagement.contentEnd + dependencyString.length + 1;
+        const range = new vscode.Range(currentDocument.positionAt(start), currentDocument.positionAt(end));
+        edit.delete(currentDocument.uri, range);
+    }
+
+    edit.insert(currentDocument.uri, insertPosition, targetText);
+    await vscode.workspace.applyEdit(edit);
+    const endingPosition: vscode.Position = currentDocument.positionAt(currentDocument.offsetAt(insertPosition) + targetText.length);
+    textEditor.revealRange(new vscode.Range(insertPosition, endingPosition));
+    vscode.workspace.saveAll();
+}
+
+function getAllVersionsInTree(pomPath: string, gid: string, aid: string): string[] {
+    const project: MavenProject | undefined = mavenExplorerProvider.getMavenProject(pomPath);
+    if (project === undefined) {
+        throw new UserError("Failed to get maven projects.");
+    }
+    const fullText: string = project.fullText;
+    const re = new RegExp(`${gid}:${aid}:[\\w.-]+`, "gm");
+    const artifacts: string[] | null = fullText.match(re);
+    let versions: string[] = [];
+    if (artifacts !== null) {
+        artifacts.forEach(a => { versions.push(a.slice(gid.length + aid.length + 2)); });
+    }
+
+    function compare(v1: string, v2: string): number {
```

Comment:
Are you sorting them in reverse order?

What's going on with below case?
`["0.1.0", "0.2.0", "0.3.0", "0.10.0"]`